app/src/main/java/org/simple/clinic/encounter/EncounterModule.kt

```diff
@@ -0,0 +1,28 @@
+package org.simple.clinic.encounter
+
+import com.f2prateek.rx.preferences2.Preference
+import com.f2prateek.rx.preferences2.RxSharedPreferences
+import dagger.Module
+import dagger.Provides
+import org.simple.clinic.encounter.sync.EncounterSyncApi
+import org.simple.clinic.util.None
+import org.simple.clinic.util.Optional
+import org.simple.clinic.util.preference.OptionalRxPreferencesConverter
+import org.simple.clinic.util.preference.StringPreferenceConverter
+import retrofit2.Retrofit
+import javax.inject.Named
+
+@Module
+class EncounterModule {
+
+  @Provides
+  fun syncApi(retrofit: Retrofit): EncounterSyncApi {
+    return retrofit.create(EncounterSyncApi::class.java)
+  }
+
+  @Provides
+  @Named("last_encounter_pull_token")
+  fun lastPullToken(rxSharedPreferences: RxSharedPreferences): Preference<Optional<String>> {
+    return rxSharedPreferences.getObject("last_encounter_pull_token_v3", None, OptionalRxPreferencesConverter(StringPreferenceConverter()))
```

Comment:
Why is there a `_v3` suffix for the pull token key? We needed this earlier because the client would need to change the token when needing to sync from the beginning, but that is unnecessary now with the server controlled resync feature.

Reply:
I kept this for consistency. I can remove it if you think it is not needed.

Reply:
I don't think it's necessary anymore, we should remove it.